File: VoxelSniperCore/src/main/java/com/github/kevindagame/util/BrushOperation/BiomeOperation.java
package com.github.kevindagame.util.BrushOperation;

import com.github.kevindagame.voxelsniper.location.VoxelLocation;

/**
 * Represents a brush operation that changes the biome of a location.
 */
public class BiomeOperation extends BrushOperation {
    private final String oldBiome;
    private String newBiome;

    public BiomeOperation(VoxelLocation location, String oldBiome, String newBiome) {
        super(location);
        this.oldBiome = oldBiome;
        this.newBiome = newBiome;
    }

    public String getOldBiome() {
        return oldBiome;
    }

    public String getNewBiome() {
        return newBiome;
    }

    public void setNewBiome(String newBiome) {
        this.newBiome = newBiome;
    }
}

File: VoxelSniperCore/src/main/java/com/github/kevindagame/util/brushOperation/operation/BiomeOperation.kt
package com.github.kevindagame.util.brushOperation.operation

import com.github.kevindagame.snipe.Undo
import com.github.kevindagame.voxelsniper.biome.VoxelBiome
import com.github.kevindagame.voxelsniper.location.BaseLocation

/**
 * Represents a brush operation that changes the biome of a location.
 */
class BiomeOperation(location: BaseLocation, val oldBiome: VoxelBiome, var newBiome: VoxelBiome) :
    BrushOperation(location) {

    override fun perform(undo: Undo): Boolean {
        val location = location
        location.world.setBiome(location.blockX, location.blockY, location.blockZ, newBiome)
        return true
    }
}